frontend/app/utils/FutureSupplier.scala

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import akka.agent.Agent
+
+import scala.concurrent.Future
+import play.api.libs.concurrent.Execution.Implicits._
+
+class FutureSupplier[T](f: => Future[T]) {
+  private val agent = Agent(f)
+
+  def refresh(): Future[T] = for {
+    refreshAlteringFuture <- agent.alter { currentRefresh =>
+      if (currentRefresh.isCompleted) f else currentRefresh
```

Comment:
Not sure that would bring a performance improvement (i.e. two transactions instead of one). And even if so, my guess is that it would be pretty negligible. But yeah, you could write something like:

```scala
currentRefresh <- agent.get
if (currentRefresh.isCompleted) {
  agent.send(_ => f)
}
currentFuture
```

Though I find it more concise and robust as it is to be honest